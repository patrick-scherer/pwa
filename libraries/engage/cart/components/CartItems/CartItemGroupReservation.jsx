import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import CardListItem from '@shopgate/pwa-ui-shared/CardList/components/Item';
import { Accordion, LocationIcon } from '../../../components';
import { i18n } from '../../../core';
import { StoreAddressOpeningHours, StoreAddressPhoneNumber } from '../../../locations';
import connect from './CartItemGroupReservation.connector';
import {
  address, addressIcon, title, accordionToggle, addressDetails,
} from './CartItemGroup.style';

/**
 * Renders the product group.
 * @param {Object} props The component props.
 * @returns {JSX.Element}
 */
const FulfillmentLocation = ({ location }) => (
  <Fragment>
    {location && (
    <CardListItem>
      <Accordion
        renderLabel={() => (
          <div className={address}>
            <div className={addressIcon}>
              <LocationIcon />
            </div>
            <div>
              <div className={title}>
                {i18n.text('locations.method.ropis')}
              </div>
              {location.name}
            </div>
          </div>
        )}
        className={accordionToggle}
      >
        <div className={addressDetails}>
          <StoreAddressOpeningHours hours={location.operationHours} />
          <StoreAddressPhoneNumber phone={address.phoneNumber} />
        </div>
      </Accordion>
    </CardListItem>
    )}
  </Fragment>
);

FulfillmentLocation.propTypes = {
  location: PropTypes.shape(),
};

FulfillmentLocation.defaultProps = {
  location: null,
};

export default connect(FulfillmentLocation);
