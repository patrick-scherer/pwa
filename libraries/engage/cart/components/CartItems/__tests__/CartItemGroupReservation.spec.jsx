import React from 'react';
import { shallow } from 'enzyme';
import CartItemGroupReservation from '../CartItemGroupReservation';

jest.mock('@shopgate/engage/components');
jest.mock('@shopgate/engage/core');
jest.mock('@shopgate/engage/locations');
jest.mock('@shopgate/pwa-ui-shared/CardList/components/Item', () => 'CardListItem');
jest.mock('../CartItem.connector', () => cmp => cmp);

describe('engage > cart > components > CartItemGroupReservation', () => {
  const location = {
    name: 'Location name',
  };

  it('should render null when no location is given', () => {
    const wrapper = shallow(<CartItemGroupReservation />);
    expect(wrapper).toBeEmptyRender();
  });

  it('should render only label', () => {
    const wrapper = shallow(<CartItemGroupReservation location={location} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render accordion', () => {
    const wrapper = shallow((
      <CartItemGroupReservation
        // eslint-disable-next-line extra-rules/no-single-line-objects
        location={{ ...location, address: { phoneNumber: '012456789' } }}
      />
    ));
    expect(wrapper).toMatchSnapshot();
    const renderLabel = wrapper.find('Accordion').prop('renderLabel');
    expect(renderLabel()).toMatchSnapshot();
  });
});
