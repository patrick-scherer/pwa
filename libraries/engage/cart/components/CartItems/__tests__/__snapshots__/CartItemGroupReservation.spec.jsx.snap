// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`engage > cart > components > CartItemGroupReservation should render accordion 1`] = `
<CardListItem>
  <Accordion
    className="css-m7h5wp"
    renderLabel={[Function]}
  >
    <div
      className={
        Object {
          "data-css-17vr7yg": "",
        }
      }
    >
      <StoreAddressOpeningHours />
      <StoreAddressPhoneNumber
        phone="012456789"
      />
    </div>
  </Accordion>
</CardListItem>
`;

exports[`engage > cart > components > CartItemGroupReservation should render accordion 2`] = `
<CartItemGroupReservationLabel
  location={
    Object {
      "address": Object {
        "phoneNumber": "012456789",
      },
      "name": "Location name",
    }
  }
/>
`;

exports[`engage > cart > components > CartItemGroupReservation should render only label 1`] = `
<CardListItem
  className="css-u4u0p5"
>
  <CartItemGroupReservationLabel
    location={
      Object {
        "name": "Location name",
      }
    }
  />
</CardListItem>
`;
